### ISSUE TITLE
Can I know which file specify your model architecture?

### ISSUE THREAD

**Issue by yxchng** [peiyunh/tiny#25]
I am not familiar with matconvnet and it seems very messy.

**Comment by chinakook** [comment]
You may try this script to draw the model architecture, It's easy!
function draw_net(net,name)
%input: net (DAGNN format)  output_filename output: jpg
%This code is modified from http://stackoverflow.com/questions/5518200/automatically-generating-a-diagram-of-function-calls-in-matlab
OutputName = 'net';
if ~isempty(name)
    OutputName = name;
end
dotFile = [OutputName '.dot'];
fid = fopen(dotFile, 'w');
fprintf(fid, 'digraph G {\n');
for i = 1:numel(net.layers)
    for j = 1:numel(net.layers(i).inputs)  % multi-input to one output
        if(iscell(net.layers(i).outputs))
            for k = 1:numel(net.layers(i).outputs)
                fprintf( fid,'%s -> %s[label = %s];\n', char(net.layers(i).inputs(j)), char(net.layers(i).outputs(k)),net.layers(i).type(7:end));
            end
            continue;
        else
            fprintf( fid,'%s -> %s', char(net.layers(i).inputs(j)), char(net.layers(i).outputs));
        end
        
        block = net.layers(i).block;
        block_class = net.layers(i).type;
        %block_class = class(block); % use this in the old version of matconvnet
        if (isequal(block_class ,'dagnn.Conv') || isequal(block_class, 'dagnn.ConvTranspose'))
            fh = block.size(1);
            fw = block.size(2);
            fprintf (fid,'[label = %s%dx%d];\n',block_class(7:end),fh,fw);
        else
            %if(~iscell(net.layers(i).outputs))
                fprintf (fid,'[label = %s];\n',block_class(7:end));
            %end
        end
    end
end
fprintf(fid, '}\n');
fclose(fid);

% Render to image
imageFile = [OutputName '.png'];
% Assumes the GraphViz bin dir is on the path; if not, use full path to dot.exe
cmd = sprintf('dot -Tpng -Gsize="256,256" "%s" -o"%s"', dotFile, imageFile);  % for better view, you can use number bigger than 32
system(cmd);
%fprintf('Wrote to %s\n', imageFile);
%im = imread(imageFile);
%imshow(im);

**Comment by peiyunh** [comment]
If you are more familiar with Caffe, there is no such prototxt file for defining the network structure. Here is an example of how you could initialize a deep network: https://github.com/vlfeat/matconvnet/blob/master/examples/imagenet/cnn_imagenet_init.m. 

In our case, since we start with pre-trained networks, it is even easier. We basically load the model and then do network surgery (if you speak Caffe's language). Such process is in https://github.com/peiyunh/tiny/blob/master/init/cnn_add_loss_fcn8s_resnet101_simple.m. 

Here is a visualization of our network: http://ethereon.github.io/netscope/#/gist/8a0d5ef37da9dc4cd611d178404b3641

**Comment by chinakook** [comment]
![tiny](https://user-images.githubusercontent.com/5417628/27093653-215aedc2-509a-11e7-9258-8e3fe27dc29a.jpg)
